diff -r ac60d4be68b7 data/Makefile.am
--- a/data/Makefile.am	Sun Jun 07 20:59:04 2009 +0200
+++ b/data/Makefile.am	Mon Jun 08 05:07:49 2009 +0300
@@ -14,13 +14,7 @@
 otherdir = $(pkgdatadir)/data/other
 other_DATA = other/servers.xml other/cacerts.pem
 
-if BUILD_REMOTE_CONTROL
-OPTIONAL_MAN = gajim-remote.1
-else
-OPTIONAL_MAN =
-endif
-
-man_MANS = gajim.1 $(OPTIONAL_MAN)
+man_MANS = gajim.1 gajim-remote.1
 
 
 EXTRA_DIST = $(desktop_in_files) \
